scripts/zkai_larger_native_boundary_candidate_selector_gate.py

```
@@ -0,0 +1,779 @@
+    return "blake2b-256:" + digest.hexdigest()
+
+
+def refresh_payload_commitment(payload: dict[str, Any]) -> None:
+    payload["payload_commitment"] = payload_commitment(payload)
+
+
+def read_json_and_raw(path: pathlib.Path, label: str) -> tuple[dict[str, Any], bytes]:
+    try:
+        raw = path.read_bytes()
+    except OSError as err:
+        raise LargerNativeBoundaryCandidateSelectorError(f"failed to read {label}: {err}") from err
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)